### DISCUSSION TITLE
First time - best practice

### DISCUSSION THREAD

**Discussion by flopp999** [keygen-sh/community#78] [Q&A] [answered]
Hi, I am new to licensing software, so please have patience :)

I have some python programs running on a computer that I want to implement a license to.

I have been playing around with all python examples but I don't understand how to do.

I have created a Policy and a License.

I have tried to use Key and also license.lic file.

I can active the license with [Machine example](https://github.com/keygen-sh/example-python-machine-activation)

So to my problem:

I do have internet access when I active machine. After one year I want the license to expire.
When it is time to reactive the license the computer doesn't have internet access.
Can I then use license or machine file to reactive the license for one more year?

thanks

**Comment by flopp999** [comment, marked as answer]
OK maybe this is the solution, I almost knew that I would maybe solve it after I open this discussion ;)

First time with internet use Machine activation, then when the license is expired, in the license click Renew and create a license.lic file, send that file to customer and run [Crypto license](https://github.com/keygen-sh/example-python-cryptographic-license-files)

is this a correct way?

**Reply by ezekg** [reply]
Yep, exactly. That's what I was going to recommend! After renewal, you could also check out [a machine file](https://github.com/keygen-sh/example-python-cryptographic-machine-files) instead of a license file. Machine files are very similar to license files, except that it's also locked to a specific machine so that it can't be shared.

**Reply by flopp999** [reply]
oh create, then I will use machine instead

Thanks
